local_groups/views.py

```diff
@@ -163,6 +156,80 @@ def post(self, request, *args, **kwargs):
             return self.redirect_user()
 
 
+class PasswordChangeView(LoginRequiredMixin, SuccessMessageMixin, FormView):
+    form_class = PasswordChangeForm
+    success_message = "Your password has been updated successfully."
+    success_url = reverse_lazy('groups-dashboard')
+    template_name = "password_change.html"
+
+    def check_old_password(self, form):
+        """Check if old password is valid in BSD"""
+        username = self.request.user.email
+        old_password = form.cleaned_data['old_password']
+        checkCredentialsResult = bsdApi.account_checkCredentials(
+            username,
+            old_password
+        )
+
+        '''
+        Should get 200 response and constituent record
+
+        https://cshift.cp.bsd.net/page/api/doc#---------------------check_credentials-----------------
```

Comment:
Random comment - the way BSD structures its API URLs is one of the more ridiculous things I've seen.